modules/rostests/apitests/kernel32/SetComputerNameExW.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for the SetComputerNameExW API
+ * COPYRIGHT:   Victor Martinez Calvo (victor.martinez@reactos.org)
+ *              Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+#include <apitest.h>
+
+#define WIN32_NO_STATUS
+#include <stdio.h>
+#include <ndk/rtltypes.h>
+
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x0600
+#include <winreg.h>
+
+static HKEY OpenHostNameKey(void)
+{
+    static const WCHAR
+        RegHostNameKey[] = L"System\\CurrentControlSet\\Services\\Tcpip\\Parameters";
+    HKEY hKey = NULL;
+    LONG Error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, RegHostNameKey, 0, KEY_ALL_ACCESS, &hKey);
+    if (!Error)
+        return hKey;
+    return NULL;
+}
+
+static HKEY OpenComputerNameKey(void)
+{
+    static const WCHAR
+        RegComputerNameKey[] = L"System\\CurrentControlSet\\Control\\ComputerName\\ComputerName";
+    HKEY hKey = NULL;
+    LONG Error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, RegComputerNameKey, 0, KEY_ALL_ACCESS, &hKey);
+    if (!Error)
+        return hKey;
+    return NULL;
+}
+
+START_TEST(SetComputerNameExW)
+{
+    LONG Error;
+    BOOL ret;
+    HKEY hKeyHN, hKeyCN;
+    DWORD cbData;
+    WCHAR szHostNameOld[MAX_PATH], szHostNameNew[MAX_PATH];
+    WCHAR szComputerNameOld[MAX_PATH], szComputerNameNew[MAX_PATH];
+    static const WCHAR szNewName[] = L"SRVROSTEST";
+
+    /* Open keys */
+    hKeyHN = OpenHostNameKey();
+    hKeyCN = OpenComputerNameKey();
+    if (!hKeyHN || !hKeyCN)
+    {
+        RegCloseKey(hKeyHN);
+        RegCloseKey(hKeyCN);
+        skip("Unable to open keys (%p, %p).\n", hKeyHN, hKeyCN);
+        return;
+    }
+
+    /* Get Old Hostname */
+    szHostNameOld[0] = UNICODE_NULL;
+    cbData = sizeof(szHostNameOld);
+    Error = RegQueryValueExW(hKeyHN, L"Hostname", NULL, NULL, (LPBYTE)szHostNameOld, &cbData);
+    ok_long(Error, ERROR_SUCCESS);
+    ok(szHostNameOld[0], "szHostNameOld is %S", szHostNameOld);
+
+    /* Get Old Computer Name */
+    szComputerNameOld[0] = UNICODE_NULL;
+    cbData = sizeof(szComputerNameOld);
+    Error = RegQueryValueExW(hKeyCN, L"ComputerName", NULL, NULL, (LPBYTE)szComputerNameOld, &cbData);
+    ok_long(Error, ERROR_SUCCESS);
+    ok(szComputerNameOld[0], "szHostNameOld is %S", szComputerNameOld);
+
+    /* Close keys */
+    Error = RegCloseKey(hKeyHN);
+    ok_long(Error, ERROR_SUCCESS);
+    Error = RegCloseKey(hKeyCN);
+    ok_long(Error, ERROR_SUCCESS);
+
+    /* Change the value */
+    ret = SetComputerNameExW(ComputerNamePhysicalDnsHostname, szNewName);
+    ok_int(ret, TRUE);
+
+    /* Open keys */
+    hKeyHN = OpenHostNameKey();
```

Comment:
Do the keys really gave to be reopened?

Reply:
Yes. Reopening is necessary.

Reply:
Interesting. Is that expected behaviour (i.e. does it work like this on Windows)?

Reply:
Without closing and reopening, it succeeded against my expectation:
![without-closing](https://user-images.githubusercontent.com/2107452/58368715-b9db7780-7f2b-11e9-8dbd-108ca39f6db4.png)
OK, I will delete reopening...

Reply:
Done.